db/change_cache.go

```diff
@@ -475,7 +475,9 @@ func (c *changeCache) DocChanged(event sgbucket.FeedEvent) {
 		nextSequence := c.getNextSequence()
 
 		for _, seq := range syncData.RecentSequences {
-			if seq >= nextSequence && seq < currentSequence {
+			// we need to check if it's possible that one of the recent sequences has been pushed to skipped
+			isSkipped := seq < currentSequence && seq >= c.getOldestSkippedSequence(ctx)
+			if (seq >= nextSequence && seq < currentSequence) || isSkipped {
```

Comment:
I think it would be helpful to document what each of these states mean. I wrote a guess but I'm not actually sure. One of the if statements ends up being duplicated in both predicates `seq < currentSequence`.

```suggestion
            // seq < currentSequence means the sequence is not the latest allocated, and is not ahead of the latest allocated (i.e. migrated in from another cluster)
            // seq >= nextSequence means this sequence is a pending sequence to be expected in a subsequent call to processEntry
            // seq >= c.getOldestSkippedSequence(ctx) means that this sequence never arrived over the caching feed and was pushed to a skipped sequence
			if seq < currentSequence && ( seq >= nextSequence || seq >= c.getOldestSkippedSequence(ctx)) {
```

`seq >= c.getOldestSkippedSequence(ctx)` acquires a read lock and while I do not think this is contentious because I doubt `recent_sequences` is large, does it make sense to move this outside the loop. Alternatively, is it wrong to move outside of the loop because there could be new skipped sequences while this is processing?

Reply:
I originally too this approach but went for this option given I wasn't totally sure at the time if it was avoidable. 

I've thought more about it today and reran test with moving the get oldest skipped call and I feel its safe to move it out of the loop.